src/ch/ricardo/plugins/intellij/mockery/action/GenMockeryActionHandler.java

```diff
@@ -98,10 +98,36 @@ public void doExecute(@NotNull Editor editor, Caret caret, DataContext dataConte
 
         // Do the generation
         Path filePath = Paths.get(FileDocumentManager.getInstance().getFile(editor.getDocument()).getCanonicalPath());
-
         GeneralCommandLine commandLine = PtyCommandLine.isEnabled() ? new PtyCommandLine() : new GeneralCommandLine();
         commandLine.withWorkDirectory(filePath.getParent().toString());
-        commandLine.setExePath(EnvironmentUtil.getValue("GOPATH") + "/bin/mockery");
+
+        String[] envGoPaths = EnvironmentUtil.getValue("GOPATH").split(File.pathSeparator);
+        String mockeryPath = "";
+
+        for (String goPath : envGoPaths) {
+            File goPathBin = new File(goPath + "/bin");
+            if(!goPathBin.exists() || !goPathBin.isDirectory())
+                continue;
+
+            String[] goPathBinFiles = goPathBin.list();
+            if(goPathBinFiles == null)
+                continue;
+
+            for(String goPathBinExecutable : goPathBinFiles)
+                if(goPathBinExecutable.startsWith("mockery.")) {
+                    mockeryPath = new File(goPath + "/bin/mockery").getAbsolutePath();
+                    break;
+                }
+
+            if(!mockeryPath.isEmpty())
+                break;
```

Comment:
I think we can remove this 'if': mockeryPath is set only in one place, and right after that place there is a break already.

Reply:
The last 'if' intented to avoid searching in the rest of the paths if mockery was already found in the current path.

Reply:
Yup, the intention is clear. What I'm saying is that if mockery was found, it means that execution passed by line 118.
And in that case, also the break at 119 is executed, so the break at 123 is probably never going to be called.

Reply:
Pay attention that the break in line 119 only breaks the nested loop (line 116). The outer loop (line 107) still going to perform.

Reply:
Ah-ha. You are right. Then please forget this one.